utils/kubernetes/describe/describe_test.go

```diff
@@ -0,0 +1,196 @@
+package describe
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	
+)
+
+// A MockClient meant to interact with the Kubernetes Api
+type MockClient struct {
+	Response      []byte //stores the response body that should be returned when the MockClient makes a request.
+	RequestUrl    string // stores the URL that the Mock client should expect when the HTTP request is performed
+	RequestMethod string // stores the HTTP request method made by the mock client
+	RequestBody   []byte // stores the request body that the fake client should expect to receive when the HTTP request is made
+	DescribeError error  //handle errors for Describe()
+
+}
+
+// the main role of the Prepend Reactor is to return httpErrors for Describerfunc
+// PrependReactor adds a reactor function that gets called before every mock request is sent.
+// The reactor function sets the HTTP status code to 404 if the Describe() function is called.
+func (m *MockClient) PrependReactor(method string, Name string, reactorFunc func(*http.Request) (*http.Response, error)) {
+	originalDo := m.Do
+
+	_ = func(req http.Request) (*http.Response, error) {
+		//call the reactor funtion for every reqeust
+		resp, err := reactorFunc(&req)
+		if err != nil || resp != nil {
+			// If the reactor function returns a response or error, return it immediately
+			return resp, err
+		}
+		// Otherwise, call the original Do() method
+		return originalDo(req)
+	}
+}
```

Comment:
I think this can be made more readable

Reply:
the <code>PrependReactor()</code> returns http Error response, if there are any errors when the DescribeFunc is called with specific options . The code here returns a reactor function for request for the DescribeFunc if the request have errors it returns the error. If it doesnt the <code>originalDo(req)</code> handles the request and returns a response